src/game_inventory.cpp

```diff
@@ -154,24 +154,100 @@ item *game::inv_map_for_liquid( const item &liquid, const std::string &title, in
         return location->get_remaining_capacity_for_liquid( liquid, allow_buckets ) > 0;
     };
 
-    return inv_map_splice( filter, title, radius,
-                           string_format( _( "You don't have a suitable container for carrying %s." ),
-                                          liquid.type_name( 1 ).c_str() ) ).get_item();
+    return inv_internal( u, inventory_filter_preset( filter ), title, radius,
+                         string_format( _( "You don't have a suitable container for carrying %s." ),
+                                        liquid.tname().c_str() ) ).get_item();
+}
+
+class gunmod_inventory_preset: public inventory_selector_preset
+{
+    public:
+        gunmod_inventory_preset( const player &p, const item &gunmod ) : p( p ), gunmod( gunmod ) {
+            append_cell( [ this ]( const item_location & loc ) -> std::string {
```

Comment:
avoid specifying return type for lambda if necessary. consistency with existing code is preferable. subjective because if you have a prevailing technical argument that precedent could be changed